src/shared/utils/index.ts

```
@@ -1,12 +1,22 @@
-import catchAsync from './catchAsync';
+import { NextFunction, Request, Response } from 'express';
 import { sendEmail } from './email';
 import { getPaginationOptions } from './pagination';
 import { isValidPhoneNumber, formatPhoneNumber } from './phoneValidator';
+import { generateRandomCode } from './codeGenerator';
+
+/**
+ * Catch async errors in controllers
+ */
+export const catchAsync = (fn: Function) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)